server/Websocket/websocketserver.mjs

```diff
@@ -159,29 +170,47 @@ function handleFrame(socket, opcode, data) {
     case OPCODE_TEXT: {
       const message = data.toString("utf8").trim();
 
-      if (message.startsWith("{") && message.endsWith("}")) {
-        try {
-          const jsonData = JSON.parse(message);
-          console.log("Received JSON:", jsonData);
-
-          const response = JSON.stringify({ message: jsonData });
-          sendFrame(socket, OPCODE_TEXT, Buffer.from(response));
-        } catch {
-          console.error("Invalid JSON:", message);
-          const errResp = JSON.stringify({
-            error: "Invalid JSON format",
-            originalMessage: message,
-          });
-          sendFrame(socket, OPCODE_TEXT, Buffer.from(errResp));
+      try {
+        const jsonData = JSON.parse(message);
+        const { action, roomName, message: msg } = jsonData;
+
+        switch (action) {
+          case "join":
```

Comment:
Fine for now, but consider moving these strings to a set of valid constants.  You can make an object like this
```
const ACTIONS = {
 JOIN: 'join',
 LEAVE: 'leave',
}
```

then use them like `ACTION.LEAVE`

What this does is prevent anyone working in the code from using invalid types, mispelling anything, etc...